x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx

```diff
@@ -289,9 +300,32 @@ export function AlertDetailsAppSection({
       }),
     },
   ];
-
   /* Error Rate */
 
+  const getLatencyChartAdditionalData = () => {
```

Comment:
Since https://github.com/elastic/kibana/pull/143298 is merged I'll add a comment about the following code in this PR:

```ts
  const transactionTypes = useServiceTransactionTypesFetcher({
    serviceName,
    start,
    end,
  });

  const transactionType = getTransactionType({
    transactionType: String(alert.fields[TRANSACTION_TYPE]),
    transactionTypes,
    agentName,
  });
```

I don't understand the above code in this context. We don't want to get a default transaction type. We want the transaction type for the rule. The transaction type is optional when creating a rule but required on alerts. This means that if the transaction type is not specified for a rule, the rule executor will do a terms agg on `transaction.type` to ensure an alert is associated with the correct transaction type. See:

https://github.com/elastic/kibana/blob/2e15fb52076895c16b6dba2e81de6ed231d09cd6/x-pack/plugins/apm/server/routes/alerts/rule_types/transaction_duration/register_transaction_duration_rule_type.ts#L179

tldr: I think the above code is redundant. Get the transaction type from the alert.

Reply:
In the same vain, why is this needed?
```ts
const latencyAggregationType = getAggsTypeFromRule(params.aggregationType);
```

`params.aggregationType` is of type `LatencyAggregationType` and the returned value is `LatencyAggregationType`.

Reply:
This also looks incorrect:
```
const environment = String(params.environment) || ENVIRONMENT_ALL.value;
```

`environment` (like `transaction.type`) is required on the alert (optional on the rule).

Reply:
@sqren, for the [environment](https://github.com/elastic/kibana/pull/147848/files#r1062486349) `const environment = String(params.environment) || ENVIRONMENT_ALL.value;
` It's the rule environment, and it was addressed in this thread https://github.com/elastic/kibana/pull/143298#discussion_r1021406794

Reply:
> I don't know the workflow and how it should be implemented.
I followed the same steps in the [APMServiceContext](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx#L69-L80) to get the tansactionType

Yes, in the UI we are fine with displaying data for a default transaction type if nothing is given. The user can always use the selector to change to another transaction type.
In this case, afaiu we are looking at an alert that already happened, right? It's not up to the user to decide what transaction type they want to use. We should show the chart for the correct transaction type, which is on the alert.

Reply:
>We want to display the latency chart for an alert so the user can understand why an alert fired, right? In that case, we should get the environment from the alert - not the rule.

@sqren, you are right. I updated it. Thanks!

Reply:
BTW, the alert env is fetched from the rule in the `createLifecycleRuleType` with some null-checks. But it's better to take from the alert.

Reply:
I still see 

```
  const transactionType = getTransactionType({
    transactionType: String(alert.fields[TRANSACTION_TYPE]),
    transactionTypes,
    agentName,
  });
```

This should be changed to:
```
const transactionType = alert.fields[TRANSACTION_TYPE];
```

Reply:
Done